app/components/announcement_banner/announcement_banner.tsx

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {
+    Text,
+    TouchableOpacity,
+    View,
+} from 'react-native';
+import Animated, {useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import {dismissAnnouncement} from '@actions/local/systems';
+import CompassIcon from '@components/compass_icon';
+import RemoveMarkdown from '@components/remove_markdown';
+import {ANNOUNCEMENT_BAR_HEIGHT} from '@constants/view';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {bottomSheet} from '@screens/navigation';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import ExpandedAnnouncementBanner from './expanded_announcement_banner';
+
+type Props = {
+    bannerColor: string;
+    bannerDismissed: boolean;
+    bannerEnabled: boolean;
+    bannerText?: string;
+    bannerTextColor?: string;
+    allowDismissal: boolean;
+}
+
+const getStyle = makeStyleSheetFromTheme((theme: Theme) => ({
+    background: {
+        backgroundColor: theme.sidebarBg,
+    },
+    bannerContainer: {
+        flex: 1,
+        paddingHorizontal: 10,
+        overflow: 'hidden',
+        flexDirection: 'row',
+        alignItems: 'center',
+        marginHorizontal: 8,
+        borderRadius: 7,
+    },
+    wrapper: {
+        flexDirection: 'row',
+        flex: 1,
+        overflow: 'hidden',
+    },
+    bannerTextContainer: {
+        flex: 1,
+        flexGrow: 1,
+        marginRight: 5,
+        textAlign: 'center',
+    },
+    bannerText: {
+        ...typography('Body', 100, 'SemiBold'),
+    },
+}));
+
+const CLOSE_BUTTON_ID = 'announcement-close';
+
+const BUTTON_HEIGHT = 48; // From /app/utils/buttonStyles.ts, lg button
+const TITLE_HEIGHT = 30 + 12; // typography 600 line height
+const MARGINS = 12 + 24 + 10; // (after title + after text + after content) from ./expanded_announcement_banner.tsx
+const TEXT_CONTAINER_HEIGHT = 150;
+const DISMISS_BUTTON_HEIGHT = BUTTON_HEIGHT + 10; // Top margin from ./expanded_announcement_banner.tsx
+
+const SNAP_POINT_WITHOUT_DISMISS = TITLE_HEIGHT + BUTTON_HEIGHT + MARGINS + TEXT_CONTAINER_HEIGHT;
+
+const AnnouncementBanner = ({
+    bannerColor,
+    bannerDismissed,
+    bannerEnabled,
+    bannerText = '',
+    bannerTextColor = '#000',
+    allowDismissal,
+}: Props) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const height = useSharedValue(0);
+    const theme = useTheme();
+    const [visible, setVisible] = useState(false);
+    const style = getStyle(theme);
+
+    const renderContent = useCallback(() => (
+        <ExpandedAnnouncementBanner
+            allowDismissal={allowDismissal}
+            bannerText={bannerText}
+        />
```

Comment:
does this actually update if the values change and the banner is already showing? it seems to be that it won't, so I guess the best option here would be to have the `ExpandedAnnouncementBanner` component to observe for the text and dismissal, unless we don't really care.

Reply:
This does not update if we are already showing. Among other things because the bottom sheet is not the best responsive part for changes (I may be misremembering, but messing with the snap points while the screen was already showing was not viable).

Reply:
I think we should err on the side of this text being shorter. It seems unlikely that a really long announcement would be entered here and the input field we use for this text is not a text area.

<img width="956" alt="image" src="https://user-images.githubusercontent.com/2040554/202213785-d0962d67-6176-4fe1-a26e-4f07752df78d.png">

Reply:
I think we should err on the side of this text being shorter. It seems unlikely that a really long announcement would be entered here and the input field we use for this text is not a text area.

<img width="956" alt="image" src="https://user-images.githubusercontent.com/2040554/202213785-d0962d67-6176-4fe1-a26e-4f07752df78d.png">